src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -23662,7 +23662,138 @@ public void Main()
+            var comp = CreateNullableCompilation(source, references: new[] { lib.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var comp2 = CreateNullableCompilation(new[] { source, lib_cs, DoesNotReturnAttributeDefinition });
+            comp2.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void DoesNotReturn_Operator()
+        {
+            // Annotations not honored on user-defined operators yet https://github.com/dotnet/roslyn/issues/32671
+            string lib_cs = @"
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [DoesNotReturn] public static C operator +(C a, C b) => throw null!;
+}
+";
+            string source = @"
+class D
+{
+    void Main(object? o, C c)
+    {
+        _ = c + c;
+        o.ToString(); // unreachable so no warning
+    }
+}
+";
+            var lib = CreateNullableCompilation(new[] { lib_cs, DoesNotReturnAttributeDefinition });
+            var comp = CreateNullableCompilation(source, references: new[] { lib.EmitToImageReference() });
+            comp.VerifyDiagnostics(
+                // (7,9): warning CS8602: Dereference of a possibly null reference.
+                //         o.ToString(); // unreachable so no warning
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "o").WithLocation(7, 9)
+                );
+
+            var comp2 = CreateNullableCompilation(new[] { source, lib_cs, DoesNotReturnAttributeDefinition });
+            comp2.VerifyDiagnostics(
+                // (7,9): warning CS8602: Dereference of a possibly null reference.
+                //         o.ToString(); // unreachable so no warning
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "o").WithLocation(7, 9)
+                );
+        }
+
+        [Fact]
+        public void DoesNotReturn_WithDoesNotReturnIf()
+        {
+            // warn on misuse of attribute (on a method that returns a value)? https://github.com/dotnet/roslyn/issues/36073
+            string lib_cs = @"
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    [DoesNotReturn] public static bool Throw([DoesNotReturnIf(true)] bool x) => throw null!;
+}
+";
+            string source = @"
+class D
+{
+    void Main(object? o, bool b)
+    {
+        _ = C.Throw(b)
+            ? o.ToString()
+            : o.ToString();
+    }
+}
+";
+            var lib = CreateNullableCompilation(new[] { lib_cs, DoesNotReturnAttributeDefinition, DoesNotReturnIfAttributeDefinition });
+            var comp = CreateNullableCompilation(source, references: new[] { lib.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var comp2 = CreateNullableCompilation(new[] { source, lib_cs, DoesNotReturnAttributeDefinition, DoesNotReturnIfAttributeDefinition });
+            comp2.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void DoesNotReturn_Ctor()
```

Comment:
>DoesNotReturn_Ctor [](start = 20, length = 18)

It looks like this test and the next are only testing the attribute definition (from the test base).

Reply:
That's true. Those two tests basically show that the `method` target doesn't include constructors and indexers.
I had those test in the test list I had brainstormed, but I agree they don't add much value. I'll remove.

---
In reply to: [297992776](https://github.com/dotnet/roslyn/pull/36810#discussion_r297992776) [](ancestors = 297992776)